kustomize/commands/edit/set/setannotation_test.go

```diff
@@ -0,0 +1,223 @@
+	}
+	return m
+}
+
+func TestRunSetAnnotation(t *testing.T) {
+	var o setAnnotationOptions
+	o.metadata = map[string]string{"owls": "cute", "otters": "adorable"}
+
+	m := makeAnnotationKustomization(t)
+	err := o.setAnnotations(m)
+	if err != nil {
+		t.Errorf("unexpected error: could not write to kustomization file")
+	}
+	// adding the same test input should work
+	err = o.setAnnotations(m)
+	if err != nil {
+		t.Errorf("unexpected error: could not write to kustomization file")
+	}
+	// adding new annotations should work
+	o.metadata = map[string]string{"new": "annotation", "owls": "not cute"}
+	err = o.setAnnotations(m)
+	if err != nil {
+		t.Errorf("unexpected error: could not write to kustomization file")
+	}
+}
+
+func TestSetAnnotationNoArgs(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	v := valtest_test.MakeHappyMapValidator(t)
+	cmd := newCmdSetAnnotation(fSys, v.Validator)
+	err := cmd.Execute()
+	v.VerifyNoCall()
+	if err == nil {
+		t.Errorf("expected an error")
+	}
+	if err.Error() != "must specify annotation" {
+		t.Errorf("incorrect error: %v", err.Error())
+	}
+}
+
+func TestSetAnnotationInvalidFormat(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	v := valtest_test.MakeSadMapValidator(t)
+	cmd := newCmdSetAnnotation(fSys, v.Validator)
+	args := []string{"exclamation!:point"}
+	err := cmd.RunE(cmd, args)
+	v.VerifyCall()
+	if err == nil {
+		t.Errorf("expected an error")
+	}
+	if err.Error() != valtest_test.SAD {
+		t.Errorf("incorrect error: %v", err.Error())
+	}
+}
+
+func TestSetAnnotationPrefixColonName(t *testing.T) {
+	var o setAnnotationOptions
+	o.metadata = map[string]string{"internal.config.kubernetes.io/options": "true"}
+
+	m := makeAnnotationKustomization(t)
+	err := o.setAnnotations(m)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err.Error())
+	}
+}
+
+func TestSetAnnotation253Prefix63Name(t *testing.T) {
+	var o setAnnotationOptions
+	o.metadata = map[string]string{`abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstu
+vwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz
+abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcde
+fghijklmnopqrstuvwxyzabcdefghijklmnopqrs/abcdefghijklmnopqrstuvwxyzabcdefghijklmnop
+qrstuvwxyzabcdefghijk`: "true"}
+
+	m := makeAnnotationKustomization(t)
+	err := o.setAnnotations(m)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err.Error())
+	}
+}
+
+func TestSetAnnotation254Prefix62Name(t *testing.T) {
+	fSys := filesys.MakeFsInMemory()
+	v := valtest_test.MakeHappyMapValidator(t)
+	cmd := newCmdSetAnnotation(fSys, v.Validator)
+	args := []string{`abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghi
+jklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmn
+opqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrs
+tuvwxyzabcdefghijklmnopqrst/abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabc
+defghij:true`}
```

Comment:
same thing here, let's do 
```
"abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghi" +
		"jklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmn" +
		"opqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrs" +
		"tuvwxyzabcdefghijklmnopqrst/abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabc" +
		"defghij:true"
```
instead of backticks to avoid the newlines.